### PR TITLE
Add optional debug_mode switch to Divvy twisted client

### PR DESCRIPTION
**Pull request by danishfoxpass**
## Description of the change

Adding an optional debug_mode switch to Twisted client to print debug information when required

## Type of change
- [ ] Bug fix
- [x] New feature

### Testing

- [x] Testing information has been added - test cases checklist and steps followed for testing.
- [x] Testing screenshot(s) attached for more information.

## Security

- [ ] This PR has security related considerations.

Tested on local
Screen Shot 2020-11-20 at 3 46 26 PM

### CODE REVIEW

File: divvy/twisted_client.py [hunk 1]
```diff
@@ -45,6 +46,8 @@ def connect(self):
         if self.connection:
             self.disconnect()
         self.connection = self._makeConnection()
+        if self.debug_mode:
+            self.log.debug('DivvyClient: {msg}', msg='Connecting to server')
```
Comment (aren) [inline divvy/twisted_client.py:50]: don't really need a substitution here

File: divvy/twisted_client.py [hunk 2]
```diff
@@ -53,6 +56,8 @@ def disconnect(self):
         if self.connection:
             self.connection.close()
             self.connection = None
+            if self.debug_mode:
+                self.log.debug('DivvyClient: {msg}', msg='Disconnected')
```
Comment (aren) [inline divvy/twisted_client.py:60]: don't really need a substitution here

File: divvy/twisted_client.py (hunk 1 above)
Comment (danishfoxpass) [reply divvy/twisted_client.py:50]: Want to check #connects and #drops and test persistence.

**Comment by danishfoxpass** [comment]
To test on 1 server for 10-20 minutes and turn off debug mode again.

**Review by aren — APPROVED** [review]

**Pull request merged**